File: 338.counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        final_arr = []

        for i in range(n+1):
            count = 0
            binary = bin(i)[2:]

            for char in binary:
                if char == "1":
                    count += 1
            final_arr.append(count)
        return final_arr

        

File: 338-counting-bits/338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans;
        for (int i=0;i<=n;i++) {
            int j = i;
            int c= 0;
            while (j > 0) {
                int k = j & 1;
                if (k) {
                    c++;
                }
                j = j >> 1;
            }
            ans.push_back(c);
        }
        return ans;
        
    }
};